packages/solid/src/reactive/scheduler.ts

```diff
@@ -49,9 +49,7 @@ function setupScheduler() {
   };
 
   if (
-    navigator &&
-    (navigator as NavigatorScheduling).scheduling &&
-    (navigator as NavigatorScheduling).scheduling.isInputPending
+    (navigator as NavigatorScheduling)?.scheduling?.isInputPending
```

Comment:
Optional chaining is actually verbose when compiled down. I don't think it's worth the change.

Reply:
Optional chaining is to help you reduce the judgment on the code, and you can do some code refactoring better. why Optional chaining is verbose when compiled down?

Reply:
>  why Optional chaining is verbose when compiled down?
Compilers does multiple steps but the code above is basically compiled as:

```js
var x, y;

if (((x = navigator.scheduling) == null ? void 0 : ((y = x.isInputPending) == null ? void 0 : y)))
```

Reply:
"compile down", means for when you make a build that targets a lower version of JS
for example ES2019 which does not support optional chaining will compile this.
```js
(navigator as NavigatorScheduling)?.scheduling?.isInputPending
```
to this:
```js
var _a;
(_a = navigator === null || navigator === void 0 ? void 0 : navigator.scheduling) === null || _a === void 0 ? void 0 : _a.isInputPending;

```

if you target a higher version of JS, then optional chaining might be the most compact.

Reply:
> if you target a higher version of JS, then optional chaining might be the most compact.
Although that's highly unlikely so in which case the previous code is actually much more build-friendly.

Reply:
thanks for your explanation :)